src/clj_jargon/permissions.clj

```diff
@@ -570,15 +585,43 @@
        cm (.getPath src) (.getPath dst) user admin-users skip-source-perms?))))
 
 (defn permissions
-  [cm user fpath]
+  [cm user fpath & {:keys [known-type] :or {known-type nil}}]
   (validate-path-lengths fpath)
-  (case (item/object-type cm fpath)
+  (case (or known-type (item/object-type cm fpath))
     :dir  (collection-perm-map cm user fpath)
     :file (dataobject-perm-map cm user fpath)
     {:read false
      :write false
      :own false}))
 
+(defn- larger-perm
+  [perm1 perm2]
+  (cond
+    (nil? perm1)
+    perm2
+
+    (nil? perm2)
+    perm1
+
+    (> (perm-order-map perm1) (perm-order-map perm2))
+    perm1
+
+    (> (perm-order-map perm2) (perm-order-map perm1))
+    perm2
+
+    :else
+    perm1))
+
+(defn- check-more-perms?
```

Comment:
I think I could get rid of this with medley's `take-upto` in the perms functions where this is passed in, but I figured I'd put up what I have first.